packages/kbn-discover-utils/src/components/app_menu/app_menu_registry.ts

```diff
@@ -81,11 +81,14 @@ export class AppMenuRegistry {
    * @param submenuId
    * @param appMenuItem
    */
-  public registerCustomActionUnderSubmenu(submenuId: string, appMenuItem: AppMenuActionCustom) {
+  public registerCustomActionUnderSubmenu(
+    submenuId: string,
+    appMenuItem: AppMenuSubmenuActionCustom | AppMenuSubmenuHorizontalRule
+  ) {
     this.customSubmenuItemsBySubmenuId.set(submenuId, [
       ...(this.customSubmenuItemsBySubmenuId.get(submenuId) ?? []).filter(
-        // prevent duplicates
-        (item) => !(item.id === appMenuItem.id && item.type === AppMenuActionType.custom)
+        // prevent duplicates and allow overrides
+        (item) => item.id !== appMenuItem.id
```

Comment:
Does this change let us override secondary submenu items too?

Reply:
Partially. Plus https://github.com/elastic/kibana/pull/195448/commits/814a654f8f6b596da4e3062ff6b40890773ba205